src/Juvix/Compiler/Concrete/Language.hs

```diff
@@ -322,8 +323,8 @@ data PatternAtom (s :: Stage)
   = PatternAtomIden (PatternAtomIdenType s)
   | PatternAtomWildcard Wildcard
   | PatternAtomEmpty Interval
-  | PatternAtomParens (PatternParensType s)
-  | PatternAtomBraces (PatternParensType s)
+  | PatternAtomParens (Maybe (SymbolType s)) (PatternParensType s)
```

Comment:
we try to follow the rule of having at most 1 argument to constructors that are not records. 
It can be refactored later, so you don't need to change it for this pr.